shell/common/shell.h

```diff
@@ -22,27 +32,47 @@
 
 namespace shell {
 
-class PlatformView;
-
-class Shell {
+class Shell final : public PlatformView::Delegate,
+                    public Animator::Delegate,
+                    public Engine::Delegate {
  public:
-  ~Shell();
+  struct ThreadPool {
```

Comment:
I agree. I think both this and `blink::Threads` are misnamed. `blink::Threads` does not actually hold thread. Instead, just the task runners. However, I cannot merge the task runners with the threads because the same thread pool (or I guess thread set) can be used to created multiple `blink::Threads` objects for the different shells.

How about `HostThreads` to hold the threads that will host the task runner and rename `blink::Threads` to `blink::TaskRunners`?

Also, we should really stop using the `blink` namespace but I was not sure if that change belonged in this patch.